azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -209,7 +209,7 @@ public int getExecutableFlows(final int projectId, final String flowId, final in
 
     final String[] hostPortSplit = hostPort.split(":");
     return this.apiGateway.callForJsonObjectMap(hostPortSplit[0],
-        Integer.valueOf(hostPortSplit[1]), "/jmx", null, paramList);
+        Integer.valueOf(hostPortSplit[1]), "/jmx", null, Optional.of(-1), paramList);
```

Comment:
Can you please use Optional.empty()? It will return false for Optional.isPresent(). Same for other places.

Reply:
Alternatively, we can use the default timeout value (`-1`) in the httpClient library and pass that when no timeout is defined.

Reply:
-1 is the default  by the library actually. So passing empty would indirectly make it -1.